internal/transform/operation.go

```diff
@@ -1082,26 +1059,38 @@ func extractOperationDetails(operation xdr.Operation, transaction ingest.LedgerT
 			details["contract_id"] = contractIdFromTxEnvelope(transactionEnvelope)
 			details["contract_code_hash"] = contractCodeHashFromTxEnvelope(transactionEnvelope)
 
-			switch args.ContractIdPreimage.Type {
-			case xdr.ContractIdPreimageTypeContractIdPreimageFromAddress:
-				fromAddress := args.ContractIdPreimage.MustFromAddress()
-				address, err := fromAddress.Address.String()
-				if err != nil {
-					panic(fmt.Errorf("error obtaining address for: %s", args.ContractIdPreimage.Type))
+			preimageTypeMap := switchContractIdPreimageType(args.ContractIdPreimage)
+			for key, val := range preimageTypeMap {
+				if _, ok := preimageTypeMap[key]; ok {
+					details[key] = val
 				}
-				details["from"] = "address"
-				details["address"] = address
-			case xdr.ContractIdPreimageTypeContractIdPreimageFromAsset:
-				details["from"] = "asset"
-				details["asset"] = args.ContractIdPreimage.MustFromAsset().StringCanonical()
-			default:
-				panic(fmt.Errorf("unknown contract id type: %s", args.ContractIdPreimage.Type))
 			}
 		case xdr.HostFunctionTypeHostFunctionTypeUploadContractWasm:
 			details["type"] = "upload_wasm"
 			transactionEnvelope := getTransactionV1Envelope(transaction.Envelope)
 			details["ledger_key_hash"] = ledgerKeyHashFromTxEnvelope(transactionEnvelope)
 			details["contract_code_hash"] = contractCodeHashFromTxEnvelope(transactionEnvelope)
+		case xdr.HostFunctionTypeHostFunctionTypeCreateContractV2:
```

Comment:
`HostFunctionTypeHostFunctionTypeCreateContractV2` is the same as `HostFunctionTypeHostFunctionTypeCreateContract` except it has `ConstructorArgs`

Reply:
for `createcontractV2` will `parameters` ever be overloaded with both function parms and contructorargs? Or are we just repurposing the `parameters_decoded` for constructor args?

Reply:
> Or are we just repurposing the parameters_decoded for constructor args?

This. It uses the same []ScVal code currently. 

> for createcontractV2 will parameters ever be overloaded with both function parms and contructorargs?

One thing to clarify is that the original parameters are for the `HostFunctionTypeHostFunctionTypeInvokeContract` invokehost operation. Whereas the `HostFunctionTypeHostFunctionTypeCreateContract` never had any parameters and now `HostFunctionTypeHostFunctionTypeCreateContractV2` has constructorArgs

So the CreateContract calls will never have "params" as they are currently exclusive to the invokeContract. And therefore won't ever be overloaded in P23.

Note also that constructorArgs won't ever be in the invokeContract. The way I see CreateContract is like CreateContract is where `ScValTypeScvLedgerKeyContractInstance` contract data entries are created. CreateContractV2 allows a single constructor function as an initialization function and has [other restrictions](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0058.md#design-rationale). Think of this like creating an object with 
```
class car():
  def __init__(self, make, model, color):
    self.make = make
    self.model = model
    self.color = color

  def accelerate(self):
    do_stuff
```
Any calls to the object functions no longer care about the __init__ params as it's already been initialized to the self.* of the object